app/scripts/components/common/mapbox/layers/vector-timeseries.tsx

```diff
@@ -0,0 +1,242 @@
+  LngLatLike,
+  VectorSourceImpl
+} from 'mapbox-gl';
+import { Feature } from 'geojson';
+import { endOfDay, startOfDay } from 'date-fns';
+import centroid from '@turf/centroid';
+
+import { requestQuickCache } from './utils';
+import { useMapStyle } from './styles';
+
+import { ActionStatus, S_FAILED, S_LOADING, S_SUCCEEDED } from '$utils/status';
+import { userTzDate2utcString } from '$utils/date';
+
+interface MapLayerVectorTimeseriesProps {
+  id: string;
+  stacCol: string;
+  date?: Date;
+  mapInstance: mapboxgl.Map;
+  sourceParams: object;
+  zoomExtent?: [number, number];
+  onStatusChange?: (result: { status: ActionStatus; id: string }) => void;
+  isHidden: boolean;
+}
+
+export function MapLayerVectorTimeseries(props: MapLayerVectorTimeseriesProps) {
+  const {
+    id,
+    stacCol,
+    date,
+    mapInstance,
+    sourceParams,
+    zoomExtent,
+    onStatusChange,
+    isHidden
+  } = props;
+
+  const theme = useTheme();
+  const { updateStyle } = useMapStyle();
+  const [featuresApiEndpoint, setFeaturesApiEndpoint] = useState('');
+
+  const minZoom = zoomExtent?.[0] ?? 0;
+
+  //
+  // Get the tiles url
+  //
+  useEffect(() => {
+    const controller = new AbortController();
+
+    async function load() {
+      try {
+        onStatusChange?.({ status: S_LOADING, id });
+        const data = await requestQuickCache({
+          url: `${process.env.API_DEV_STAC_ENDPOINT}/collections/${stacCol}`,
+          method: 'GET',
+          controller
+        });
+
+        setFeaturesApiEndpoint(data.links.find((l) => l.rel === 'child').href);
+        onStatusChange?.({ status: S_SUCCEEDED, id });
+      } catch (error) {
+        if (!controller.signal.aborted) {
+          setFeaturesApiEndpoint('');
+          onStatusChange?.({ status: S_FAILED, id });
+        }
+        return;
+      }
+    }
+
+    load();
+
+    return () => {
+      controller.abort();
+    };
+  }, [mapInstance, id, stacCol, date, onStatusChange]);
+
+  //
+  // Generate Mapbox GL layers and sources for raster timeseries
+  //
+  const haveSourceParamsChanged = useMemo(
+    () => JSON.stringify(sourceParams),
+    [sourceParams]
+  );
+  useEffect(() => {
+    if (!date || !featuresApiEndpoint) return;
+
+    const start = userTzDate2utcString(startOfDay(date));
+    const end = userTzDate2utcString(endOfDay(date));
+
+    const tileParams = qs.stringify({
+      ...sourceParams,
+      datetime: `${start}/${end}`
+    });
+
+    const sources: Record<string, AnySourceImpl> = {
+      [id]: {
+        type: 'vector',
+        tiles: [`${featuresApiEndpoint}/tiles/{z}/{x}/{y}?${tileParams}`]
+      } as VectorSourceImpl
+    };
+
+    const layers = [
+      {
+        id: `${id}-line-bg`,
+        type: 'line',
+        source: id,
+        'source-layer': 'default',
+        layout: {
+          visibility: isHidden ? 'none' : 'visible'
+        },
+        paint: {
+          'line-color': theme.color?.['base-300'],
+          'line-width': 4
```

Comment:
This could probably use some zoom-dependent styling, for instance:
```suggestion
          'line-width': [
              "interpolate",
              ["linear"],
              ["zoom"],
              minZoom,
              4,
              zoomExtent?.[1],
              10
            ]
```